tools/parallelize/parallelize.cpp

```diff
@@ -171,6 +190,12 @@ extern "C" int wmain(int argc, wchar_t* argv[]) {
                     continue;
                 }
 
+                if (!binary_search(

```

Comment:
binary_search and needing is_sorted checks and similar really seems suboptimal when N is 4. It will always do lg(4)+1==3 comparisons -- and those are relatively expensive < comparisons rather than == comparisons that don't get to short circuit on size -- while linear search on average will do 2 comparisons assuming the input is uniformly distributed.

Proposal (also might save a memory allocation):

```
if (!any_of(accepted_extensions.begin(), accepted_extensions.end(),
    [&] (auto& extension) { return dirEntry.path().native().ends_with(extension); }) {
    printf("Skipping non-accepted-extension %ls\n", dirEntry.path().c_str());
    continue;
}
```

Reply:
I guess in my attempt to avoid memory allocation I broke the size thing but still :)

Reply:
Optimizing the lookups when we’re doing a filesystem traversal seems unnecessary. You have a great point about the `==` size checks, but your proposed transformation is incorrect - `”meow.txt”` ends with `””`.

I think the code as written is clear and consistent with `validate.cpp`; would you be fine with wontfixing this?

Reply:
Right; it's just odd to do binary_search in a place where perf doesn't matter :P

Reply:
Developer laziness - I had copy-pasted this from `validate.cpp` where the list of skipped directories, etc. might grow indefinitely, although this one won’t.

Reply:
Ah, I see. Carry on then!